bundles/http/Citrix/Employee-Survey/script.js

```diff
@@ -0,0 +1,153 @@
+    let offset = 0
+    let total = 0
+    do {
+        const itemsResponse = await client.fetch(`item/app/${integrationParameters.surveyAdminAppId}/filter`, {
+            method: "POST",
+            body: JSON.stringify({
+                limit: limit,
+                offset: offset,
+                sort_by: "app_item_id",
+                sort_desc: true
+            })
+        })
+        const { responseBody } = await validateResponse(itemsResponse)
+        total = responseBody.total
+        for (const item of responseBody.items ?? []) {
+            const launchDate = getDate(item?.fields ?? [], 'Launch Date') ?? null
+            const responseAppId = getTextValue(item?.fields ?? [], 'text', 'Response App Id') ?? null
+            const name = getTextValue(item?.fields ?? [], 'text', 'Survey Name')
+            const webFormId = responseAppId !== null ? await getAppWebFormIds(client, responseAppId, name) : null
+            dataStore.save('survey_app_data', {
+                "response_app_id": parseInt(responseAppId),
+                "app_item_id": item.item_id,
+                "name": name,
+                "status": getCategoryData(item?.fields ?? [], 'Status'),
+                "notification_date": launchDate !== null ? moment(launchDate).subtract(1, 'd').format('YYYY-MM-DD') : null,
+                "launch_date": launchDate,
+                "expire_after": getDate(item?.fields ?? [], 'Expire After'),
+                "webform_link": responseAppId !== null ? `https://podio.com/webforms/${responseAppId}/${webFormId.form_id}` : null,
+                "survey_description": getTextValue(item?.fields ?? [], 'text', 'Survey Description'),
+            })
+        }
+        offset += limit
+    } while (dataUpdateAfterAction !== true && offset < total)
+}
+
+async function getAppWebFormIds(client, responseAppId, surveyName) {
+    const webForm = {}
+    const appWebFormsResponse = await client.fetch(`form/app/${responseAppId}`)
+    const { responseBody } = await validateResponse(appWebFormsResponse, surveyName);
+    (responseBody ?? []).forEach(app => {
+        webForm.form_id = app?.form_id ?? null
+    })
+    return webForm
+}
+
+async function updateSurvey(param) {
+    const { client, actionParameters } = param
+    let fields = {
+        "launch-date": {
+            "start": moment(actionParameters.launch_date).format('YYYY-MM-DD HH:mm:ss')
+        },
+        "status": actionParameters.status,
+    }
+    if (actionParameters.expire_after != null) {
+        fields['expire-after'] = {
+            "start": moment(actionParameters.expire_after).format('YYYY-MM-DD HH:mm:ss')
+        }
+    }
+    const updateRecordResonse = await client.fetch(`/item/${actionParameters.app_item_id}`, {
+        method: 'PUT',
+        body: JSON.stringify({
+            fields: fields
+        })
+    })
+    await validateResponse(updateRecordResonse)
+    await getSurveyData(param, true)
+}
+
+function getDate(dataItems, label) {
```

Comment:
```suggestion
function getDate(dataItems = [], label) {
```
Now you don't have to do fallback to `[]` on lines 87, 98, ...